src/main/scala/com/fulcrumgenomics/util/ProgressLogger.scala

```diff
@@ -42,9 +46,43 @@ case class ProgressLogger(logger: Logger,
   /** Method to use to record progress when genome location isn't available or relevant. */
   def record(): Boolean = record(null, -1)
 
-  def record(rec: SamRecord): Boolean = record(rec.asSam)
+  def record(rec: SamRecord): Boolean = super.record(rec.asSam)
+
+  def record(variant: Variant): Boolean = record(variant.chrom, variant.pos)
+
+  def record(interval: Interval): Boolean = record(interval.getContig, interval.getStart)
 
   /** Logs the last record if it wasn't already logged. */
   def logLast(): Boolean = super.log() // Calls the super's log() method, not log(message: String*)
+}
+
+object ProgressLogger {
 
+  /** Provides a method to record items of the given type.  If [[ItemType]] is not one of [[SamRecord]],
+    * `(String, Int)`, [[Variant]], or [[Interval], [[ProgressLogger.record()]] will be used. */
+  sealed abstract class ProgressLoggingHelper[ItemType: TypeTag] {
+    protected val record: (ProgressLogger, ItemType) => Boolean = {
+      typeOf[ItemType] match {
+        case t if t =:= typeOf[SamRecord]     => (p: ProgressLogger, item: ItemType) => p.record(item.asInstanceOf[SamRecord])
+        case t if t =:= typeOf[(String, Int)] => (p: ProgressLogger, item: ItemType) => { val (contig, pos) = item.asInstanceOf[(String, Int)]; p.record(contig, pos) }
+        case t if t =:= typeOf[Variant]       => (p: ProgressLogger, item: ItemType) => p.record(item.asInstanceOf[Variant])
+        case t if t =:= typeOf[Interval]      => (p: ProgressLogger, item: ItemType) => p.record(item.asInstanceOf[Interval])
```

Comment:
I thnk `=:=` require equality of type right?  Perhaps you want to use `t <:< typeOf[Interval]` that way any sub-type of Interval can also be handled?

Reply:
And in fact, then you could use `Locatable` instead of `Interval` and catch a lot more stuff.